inc/planningcsv.class.php

```diff
@@ -0,0 +1,187 @@
+   private $quote = '"';
+   private $titles = [
+      'actor',
+      'title',
+      'itemtype',
+      'items_id',
+      'begindate',
+      'enddate'
+   ];
+   private $lines = null;
+   private $filename = 'planning.csv';
+
+   /**
+    * @param integer $who          user ID
+    * @param integer $who_group    group ID
+    * @param string $limititemtype itemtype only display this itemtype (default '')
+   **/
+
+   public function __construct($who, $who_group = null, $limititemtype = '') {
+      $this->users_id = $who;
+      $this->groups_id = $who_group;
+      $this->limititemtype = $limititemtype;
+   }
+
+   /**
+    * Build CSV lines
+    *
+    * @return void
+    */
+   private function buildLines() {
+      global $CFG_GLPI;
+
+      $interv = [];
+      $this->lines = [];
+      $begin  = time()-MONTH_TIMESTAMP*12;
+      $end    = time()+MONTH_TIMESTAMP*12;
+      $begin  = date("Y-m-d H:i:s", $begin);
+      $end    = date("Y-m-d H:i:s", $end);
+      $params = [
+         'who'       => $this->users_id,
+         'who_group' => $this->groups_id,
+         'whogroup'  => $this->groups_id,
+         'begin'     => $begin,
+         'end'       => $end
+      ];
+
+      if (empty($this->limititemtype)) {
+         foreach ($CFG_GLPI['planning_types'] as $itemtype) {
+            $interv = array_merge($interv, $itemtype::populatePlanning($params));
+         }
+      } else {
+         $interv = $this->limititemtype::populatePlanning($params);
+      }
+
+      if (count($interv) > 0) {
+         foreach ($interv as $val) {
+            $dateBegin = new DateTime($val["begin"]);
+            $dateBegin->setTimeZone(new DateTimeZone('UTC'));
+
+            $dateEnd = new DateTime($val["end"]);
+            $dateEnd->setTimeZone(new DateTimeZone('UTC'));
+
+            $itemtype = new $val['itemtype'];
+            //(acteur;titre item;id item;date-heure début,date-heure fin;catégorie)
+            $this->lines[] = [
+               'actor'     => $val['users_id'],
+               'title'     => $val['name'],
+               'itemtype'  => $itemtype->getType(),
+               'items_id'  => $val[$itemtype->getForeignKeyField()],
+               'begindate' => $dateBegin->format('Y-m-d H:i:s'),
+               'enddate'   => $dateEnd->format('Y-m-d H:i:s')
+            ];
+         }
+      }
+   }
+
+   /**
+    * Outputs CSV export
+    *
+    * @param boolean $text Outputs text only, without headers
+    */
+   public function output($text = false) {
+      $this->getlines();
+
+      if ($text === false) {
+         header("Expires: Mon, 26 Nov 1962 00:00:00 GMT");
+         header('Pragma: private'); /// IE BUG + SSL
+         header('Cache-control: private, must-revalidate'); /// IE BUG + SSL
+         header("Content-disposition: filename=".$this->filename);
+         header('Content-type: application/octetstream');
+         // zero width no break space (for excel)
+      }
+
+      echo"\xEF\xBB\xBF";
```

Comment:
maybe add a constant for bom

Reply:
This was copy/pasted from Search; but I'm not sure this is really required, I've dropped for now